### DISCUSSION TITLE
How can I batch convert a big collection of data URIs into images?

### DISCUSSION THREAD

**Discussion by ncvbn** [ImageMagick/ImageMagick#2398] [Help]
I have a big collection of data URIs in a text file. I'd like to batch convert them into a collection of image files. What's the easiest way to do that? Thanks!

**Comment by snibgo** [comment]
What version of IM, on what platform?

What is a "data URI"? How do you make an image file from a data URI? Do you want each image to be a rasterized version of the text (in which case it doesn't matter what the text means)?

**Comment by ncvbn** [comment]
I'm on Windows 7. Here's a site that converts a data URI to an image (one at a time, not batch): https://www.site24x7.com/tools/datauri-to-image.html

I don't think rasterization has anything to do with it. Thanks!

**Comment by fmw42** [comment]
What format is your text file? Is it a .txt or .doc or .html? Please post an example. What is a data URI? Is that an embedded image? Is it a link to an external image?

**Comment by snibgo** [comment]
This is what ImageMagick calls the "inline:" format. If you have a text file x.txt that contains an inline image, you can read it just like any image, but you need the "inline:" prefix:

```
magick inline:x.txt out.png
```
To convert multiple files, use mogrify or a shell loop.

**Comment by pourpc** [comment]
You may do that by using some online url to text converters and then from text to image converters. Just google it and your work will be done right away.
